monitoring/system_pods.go

```diff
@@ -0,0 +1,218 @@
+	for _, pod := range pods {
+		if err := verifyPodStatus(pod.Status); err != nil {
+			reporter.Add(systemPodsFailureProbe(r.Name(), pod.Name, pod.Namespace))
+		}
+	}
+}
+
+// verifyPodStatus verifies the status phase and conditions.
+func verifyPodStatus(status corev1.PodStatus) error {
+	// https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
+	switch status.Phase {
+	case corev1.PodPending, corev1.PodSucceeded:
+		return nil
+	case corev1.PodRunning:
+		return trace.Wrap(verifyConditions(status.Conditions))
+	case corev1.PodFailed:
+		return trace.BadParameter("pod has failed")
+	case corev1.PodUnknown:
+		log.Warn("Pod is in unknown state.")
+		return nil
+	default:
+		log.WithField("phase", status.Phase).Warn("Pod is in invalid phase.")
+		return trace.BadParameter("pod is in invalid phase")
+	}
+}
+
+// verifyConditions verifies all pod conditions.
+func verifyConditions(conditions []corev1.PodCondition) error {
+	for _, condition := range conditions {
+		if err := verifyCondition(condition); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
+// verifyCondition verifies the provided pod condition.
+// The pod is expected to have `Initialized` and`PodScheduled` set to true.
+// The pod does not need to have `Ready` or `ContainersReady` set to true.
+func verifyCondition(condition corev1.PodCondition) error {
+	// https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions
+	switch condition.Type {
+	case corev1.PodScheduled, corev1.PodInitialized:
+		return trace.Wrap(verifyConditionIsTrue(condition))
+	case corev1.PodReady, corev1.ContainersReady:
+		return nil // `Running` pods are not always expected to be `Ready`. e.g. `gravity-site`.
+	default:
+		log.WithField("condition", condition.Type).Warnf("Received invalid pod condition.")
+		return trace.BadParameter("received invalid pod condition: %s", condition.Type)
+	}
+}
+
+// verifyConditionIsTrue verifies that the provided condition status is true.
+func verifyConditionIsTrue(condition corev1.PodCondition) error {
+	switch condition.Status {
+	case corev1.ConditionTrue:
+		return nil
+	case corev1.ConditionFalse:
+		return trace.BadParameter("%s condition is false; reason: %s", condition.Type, condition.Reason)
+	case corev1.ConditionUnknown:
+		log.Warnf("%s condition is in an unknown state.", condition.Type)
+		return nil
+	default:
+		log.WithField("status", condition.Status).Warnf("%s condition is in an invalid state", condition.Type)
+		return trace.BadParameter("%s condition is in an invalid state: %s", condition.Type, condition.Status)
+	}
+}
+
+// systemPodsFailureProbe constructs a probe that represents a failed system pods
+// check for the pod specified by podName and namespace.
+func systemPodsFailureProbe(checkerName, podName, namespace string) *pb.Probe {
+	return &pb.Probe{
+		Checker:  checkerName,
+		Detail:   fmt.Sprintf("pod %s running in namespace %s is in an invalid state", podName, namespace),
```

Comment:
I think this is missing the actual pod state (e.g. CrashLoopBackOff, Error, etc.). Can this be added? I'd also make the message a bit shorter e.g.

```
fmt.Sprintf("pod %v/%v is not Running: %v", namespace, name, <actual state>)
```